### DISCUSSION TITLE
How to make constrain with sub-matrix and it's rank

### DISCUSSION THREAD

**Discussion by bioinfor-123** [yalmip/YALMIP#1340] [Problems using YALMIP]
There is a 10000*5000 constant matrix P, x=binvar(10000,1), y=binvar(5000,1), sub matrix Q is from P, i.e. leave the corresponding rows and columns intact according to x(i)=1 and y(j)=1, then compute out r=rank(Q). I want to make a constraint with [r==50], how should I do with Yalmip ?

**Comment by johanlofberg** [comment]
You would first have to express that as a suitable optmization problem, which I think is hard as rank is a very nasty thing to express
